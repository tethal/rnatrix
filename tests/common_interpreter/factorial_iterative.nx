fun factorial(n) {
    var result = 1;
    var i = 2;
    while (i <= n) {
        result = result * i;
        i = i + 1;
    }
    return result;
}

fun main() {
    print(factorial(0));
    print(factorial(1));
    print(factorial(5));
    print(factorial(10));
    print(factorial(12));
}