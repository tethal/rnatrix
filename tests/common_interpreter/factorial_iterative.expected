1
1
120
3628800
479001600
